tests/jballerina-unit-test/src/test/resources/test-src/bala/test_bala/types/public_and_private_types_test.bal

```diff
@@ -24,3 +24,9 @@ function testModulePublicAndPrivateTypes() {
         test:assertEquals(timeZone.getInt({year:1, month: 2, day: 3}), 6);
     }
 }
+
+function testAnonymousDistinctErrorTypes() {
+    error err = error("");
+    boolean ans = err is test_module:InvalidDocumentError;
+    test_module:InvalidDocumentError|error err2 = trap <test_module:InvalidDocumentError>err;
```

Comment:
Can we add an assertion also? Maybe a positive and negative case that will help assert that the type is correct also.

Reply:
Will do this.